### PR TITLE
Add sorting query option on index controller.

### PR DESCRIPTION
**Pull request by rolandnsharp**
Tested.

### CODE REVIEW

File: test/sort-test.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                sorted.forEach(function(number, index){
+                    if (index > 1) {
+                        (sorted[index-1] <= number).should.be.true;
+                    }
+                })
+                done();
+            });
+    });
+
+    it.only('should sort documents in descending order by number.', function(done) {
```
Comment (tonymilne) [inline test/sort-test.js]: `.only` should be removed.

**Comment by tonymilne** [comment]
Similar to a previous discussion about writing good tests; try to write your test statements to assert properties of the feature, not specifics of your test case.
E.g. "should sort documents in descending order by number." is leaking your specific test case details (e.g. *by number*). Does this mean the sorting only works for number?

File: lib/pagination.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ module.exports = {
 				// http://grokbase.com/t/gg/mongoose-orm/12bwde7vj4/mongoose-executing-a-query-twice-once-with-count-and-one-for-the-dataset
 				query
 				.find()
+				.sort(options.sort && options.sort.replace('+', ' '))
```
Comment (willmcclellan) [inline lib/pagination.js]: Not 100% on the use of plus as the chaining character. It's a bit confusing when used alongside - for desc… Is comma (,) a better choice?

Comment (willmcclellan) [inline lib/pagination.js]: As mentioned offline we can't use comma… Maybe + is the right choice (its used elsewhere) just looked odd next to -

File: test/sort-test.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    it('should sort by "field" ascending.', function(done) {
+        var handler = crud.index();
+        app.get('/', handler, function(err, req, res, next) {
+            done();
+
+        });
+
+        request(app)
+            .get('/?sort=number')
```
Comment (willmcclellan) [inline test/sort-test.js]: You can do this using the query option of supertest

```
request(app)
   .get('/')
   .query({ sort: number })
```

File: lib/pagination.js (hunk 2 above)
Comment (tonymilne) [inline lib/pagination.js]: What do other popular libraries / frameworks use?

Comment (willmcclellan) [inline lib/pagination.js]: JSON API actually recommends comma for this purpose http://jsonapi.org/format/#fetching-sparse-fieldsets but it goes against what's in the spec http://stackoverflow.com/questions/14663813/using-commas-in-urls-can-break-the-url-sometimes

I knew I'd used commas in URLs before! Maybe it's more a recommendation than requirement

**Comment by willmcclellan** [comment]
:+1:

**Pull request merged**